### PR TITLE
docs: add `overrides` and `jobManifestPath` to verify docs

### PR DESCRIPTION
**Pull request by maggieneterval**
Fixes: https://github.com/GoogleContainerTools/skaffold/issues/8743

**Description**
First pass at improving `skaffold verify` documentation:
* Clarify the difference between local and Kubernetes execution modes.
* Specify how to customize Kubernetes test Jobs using the new `overrides` and `jobManifestPath` options.

### CODE REVIEW

**Comment by maggieneterval** [comment]
@dorbin PTAL - thanks!

**Comment by maggieneterval** [comment]
@dorbin and @aaron-prindle https://github.com/GoogleContainerTools/skaffold/issues/8743 also requests that `jobManifestPath` and `overrides` be included in the YAML schema documentation, but I am already seeing them show up for https://skaffold.dev/docs/references/yaml/?version=v4beta4#verify and https://skaffold.dev/docs/references/yaml/?version=v4beta5. That said, they don't show up for the default (latest stable?) version of the YAML schema page. Wanted to confirm that was expected. Thanks!

File: docs-v2/content/en/samples/verify/verify.yaml [hunk 1]
```diff
@@ -12,13 +12,15 @@ verify:
     name: integration-test-container
     image: integration-test-container
   executionMode:
-    kubernetesCluster: {}
+    kubernetesCluster:
```
Comment (aaron-prindle) [inline docs-v2/content/en/samples/verify/verify.yaml:15]: Can this be commented out w/ some indication the field is optional?

File: docs-v2/content/en/samples/verify/verify.yaml [hunk 2]
```diff
@@ -12,13 +12,15 @@ verify:
...
+      overrides: '{ "apiVersion": "v1", "spec": { ... } }'
 - name: metrics-test-container
   container:
     name: metrics-test-container
     image: metrics-test-container
   executionMode:
-    kubernetesCluster: {}
+    kubernetesCluster:
```
Comment (aaron-prindle) [inline docs-v2/content/en/samples/verify/verify.yaml:22]: Can this be commented out w/ some indication the field is optional?

File: docs-v2/content/en/samples/verify/verify.yaml (hunk 1 above)
Comment (maggieneterval) [reply docs-v2/content/en/samples/verify/verify.yaml]: Good idea. I made a couple changes to help clarify that these fields are optional:
1. Changed wording on line 28 to "There are two ways to optionally customize the Skaffold-generated Kubernetes Job"
2. Added "optional" to describe both override configuration options on lines 37 and 38
3. Commented out the optional lines in the YAML and added "[optional]" above to further clarify.
Let me know what you think!

File: docs-v2/content/en/samples/verify/verify.yaml (hunk 2 above)
Comment (maggieneterval) [reply docs-v2/content/en/samples/verify/verify.yaml]: See response above.

**Review by aaron-prindle — APPROVED** [review]
LGTM!

**Review by dorbin — COMMENTED** [review]
Looks good, Maggie! Thanks!

**Pull request merged**